File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text:")
sentences =  text.count('.') + text.count('!') + text.count('?') + text.count('...') #counting all the pontuation types
words = text.count(' ') + 1 #getting the number of words
noalpha = 0
for i in range(0 , len(text)):
    if text[i].isalpha() == False:
        noalpha += 1
letters = len(text) - noalpha #the rest should be counted as letters
letters = letters * (100/words) #putting per 100 words
sentences = sentences * (100/words) #putting per 100 words
index = round((0.0588 * letters) - (0.296 * sentences) - 15.8) #the Colmena-Liau index
#printing results
if (index < 1):
    print("Before Grade 1")
elif (index > 16 ):
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: exercises/pset1/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
int main(void)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;

    string s = get_string("Type a text: ");
    for (int i = 0, f = strlen(s); i < f; i++)
    {
        //counting letters
        if (isupper(s[i]) + islower(s[i]) != 0)
        {
            letters += 1;
        }
        else;
        //counting words
        if (isspace(s[i]) != 0)
        {
            words += 1;
        }
        else;
        //counting sentences
        if (s[i] == 33 || s[i] == 46 || s[i] == 63)
        {
            sentences += 1;
        }
    }
    //changing proportionally to 100 words
    float L = letters * (float)100 / words;
    float S = sentences * (float)100 / words;
    //applying formula
    int Grade = round((0.0588 * L) - (0.296 * S) - 15.8);
    if (Grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else    if (Grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", Grade);
    }
}